Dockerfile

```diff
@@ -30,66 +28,50 @@ RUN apt install -y --no-install-recommends openmpi-bin openssh-server openssh-cl
 
 # Build and install MPI File Utils and all dependencies
 
-RUN mkdir -p /deps
+RUN mkdir -p /deps /mfu
 WORKDIR /deps
 
 RUN wget https://github.com/hpc/libcircle/releases/download/v0.3/libcircle-0.3.0.tar.gz \
     && tar xfz libcircle-0.3.0.tar.gz \
     && cd libcircle-0.3.0 \
-    && ./configure --prefix=/usr/lib \
+    && ./configure --prefix=/deps/libcircle/lib \
+    && make install
+
+RUN wget https://github.com/libarchive/libarchive/releases/download/v3.5.3/libarchive-3.5.3.tar.gz \
+    && tar xfz libarchive-3.5.3.tar.gz \
+    && cd libarchive-3.5.3 \
+    && ./configure --prefix=/deps/libarchive/lib \
     && make install
 
 RUN wget https://github.com/llnl/lwgrp/releases/download/v1.0.3/lwgrp-1.0.3.tar.gz \
     && tar xfz lwgrp-1.0.3.tar.gz \
     && cd lwgrp-1.0.3 \
-    && ./configure --prefix=/usr/lib \
+    && ./configure --prefix=/deps/lwgrp/lib \
     && make install
 
 RUN wget https://github.com/llnl/dtcmp/releases/download/v1.1.1/dtcmp-1.1.1.tar.gz \
     && tar xfz dtcmp-1.1.1.tar.gz \
     && cd dtcmp-1.1.1 \
-    && ./configure --prefix=/usr/lib --with-lwgrp=/usr/lib \
-    && make install
-
-RUN wget https://github.com/libarchive/libarchive/releases/download/v3.5.3/libarchive-3.5.3.tar.gz \
-    && tar xfz libarchive-3.5.3.tar.gz \
-    && cd libarchive-3.5.3 \
-    && ./configure --prefix=/usr/lib \
+    && ./configure --prefix=/deps/dtcmp/lib --with-lwgrp=/deps/lwgrp/lib \
     && make install
 
 ARG MPI_FILE_UTILS_VERSION="0.11"
 RUN wget https://github.com/hpc/mpifileutils/archive/v${MPI_FILE_UTILS_VERSION}.tar.gz \
     && tar xfz v${MPI_FILE_UTILS_VERSION}.tar.gz \
     && mkdir build \
     && cd build \
-    && cmake ../mpifileutils-${MPI_FILE_UTILS_VERSION} -DWITH_DTCMP_PREFIX=/usr/lib -DWITH_LibCircle_PREFIX=/usr/lib -DWITH_LibArchive_PREFIX=/usr/lib -DCMAKE_INSTALL_PREFIX=/usr \
+    && cmake ../mpifileutils-${MPI_FILE_UTILS_VERSION} \
+        -DWITH_LibCircle_PREFIX=/deps/libcircle/lib \
+        -DWITH_DTCMP_PREFIX=/deps/dtcmp/lib \
+        -DWITH_LibArchive_PREFIX=/deps/libarchive/lib \
+        -DCMAKE_INSTALL_PREFIX=/mfu \
     && make install
 
-RUN rm -rf /deps
-
-# The following several lines are from the mpi-operator base Dockerfile
-# https://github.com/kubeflow/mpi-operator/blob/master/build/base/Dockerfile
-
-ARG port=2222
-
-# Add priviledge separation directoy to run sshd as root.
-RUN mkdir -p /var/run/sshd
-# Add capability to run sshd as non-root.
-RUN setcap CAP_NET_BIND_SERVICE=+eip /usr/sbin/sshd
+FROM mpioperator/openmpi
```

Comment:
Yes, from here on down is awesome, and the only part that really matters.
Nice.